editor/src/clj/editor/scene_visibility.clj

```diff
@@ -254,6 +255,16 @@
              (hide-outline-name-paths! scene-visibility (g/node-value scene-visibility :selected-hideable-outline-name-paths))
              (show-outline-name-paths! scene-visibility (g/node-value scene-visibility :selected-showable-outline-name-paths)))))))
 
+(handler/defhandler :hide-toggle :workbench
+  (active? [scene-visibility evaluation-context user-data]
+           (g/node-value scene-visibility :active-scene-resource-node evaluation-context))
+  (run [scene-visibility user-data]
+       (let [{:keys [node-outline-key-path]} user-data
+             outline-name-paths #{(vec (rest node-outline-key-path))}]
+         (if (contains? (g/node-value scene-visibility :hidden-node-outline-key-paths) node-outline-key-path)
+           (show-outline-name-paths! scene-visibility outline-name-paths)
+           (hide-outline-name-paths! scene-visibility outline-name-paths)))))
+
```

Comment:
A new handler was introduced to toggle the visibility state of a single nodes. The existing context menu action is applied to all selected elements and its children. That might result in unexpected behavior, so I decided to deviate. For instance, if we try to unhide a node that has at least one visible children using the context menu, the child will turn invisible, instead of changing the visibility of the selected parent.